### DISCUSSION TITLE
Detector definition with variable pixel geometry

### DISCUSSION THREAD

**Discussion by djm87** [silx-kit/pyFAI#2057] [Q&A]
Hi, 

I have a sensor with different pixel sizes (e.g. at edge vs interior). In the detector definition (which is a group of these sensors) is it a good idea to define custom pixels (i.e. an array of pixel corners of size(Ny, Nx, Nc, 3)) or will this have some unwanted side effects during integration? Is there an example that I can work from?

Thanks, 
Dan

**Comment by kif** [comment]
Hi Dan,

Most pixelated detectors (Pilatus, Eiger, maxipix, xpad, ...) have larger pixels on the border in comparison to the other pixels. Some manufacturers (Dectris) extrapolate the signal to a fully regular grid ... look the variance of an image and you find easily which pixels have been "photoshopped". 
Other manufacturers just describe this, and the analysis software needs to cope with it:
for imXpad the border pixels are 2.5x larger. You can get inspiration from this class:
https://github.com/silx-kit/pyFAI/blob/main/src/pyFAI/detectors/_imxpad.py#L56

Finally, in pyFAI, you can define the position of every single pixel corner, so each pixel could have whatever size. This detector definition has to be saved into a hdf5/nexus-format file... but it is the least easy way of doing things.

So, to answer your question, no side effect is expected ...

Cheers,

Jerome
